core/src/nucleus/ribosome/api/call.rs

```diff
@@ -139,32 +135,8 @@ fn bridge_call(runtime: &mut Runtime, input: ZomeFnCallArgs) -> Result<JsonStrin
                 Err(_) => string_response,
             };
             // Below, sanitized_response is the same response but guaranteed without quotes.
-
-            // Now we unwrap the Result by parsing it into a Result<Value, _>:
-            let maybe_parsed_result: Result<
-                Result<Value, HolochainError>,
-                serde_json::error::Error,
-            > = serde_json::from_str(&sanitized_response);
-            maybe_parsed_result
-                .map_err(|e| HolochainError::SerializationError(e.to_string()))
-                .and_then(|result| match result {
-                    Ok(value) => Ok(JsonString::from(value)),
-                    Err(error) => Err(error),
-                })
-                .or_else(|_| {
-                    // If we could not parse the result into a Result, well then the called
-                    // zome function maybe returns a plain return value.
-                    // We don't want to make this impossible so we just return that as
-                    // JsonString.
-                    // We just need to make sure it is valid JSON, so we do try to parse
-                    // one last time:
-                    let maybe_parsed_value: Result<Value, serde_json::error::Error> =
-                        serde_json::from_str(&sanitized_response);
-                    match maybe_parsed_value {
-                        Ok(parsed_value) => Ok(JsonString::from(parsed_value)),
-                        Err(_) => Ok(JsonString::from(RawString::from(sanitized_response))),
-                    }
-                })
+            // This should be returned as a JsonString for handling in the zome code.
+            Ok(JsonString::from_json(&sanitized_response))
```

Comment:
This is the main change to core. The unwrapping code is dropped and the JsonString result returned directly as for a zome call